src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/OpenProjectsHandler.cs

```diff
@@ -17,18 +18,22 @@ internal class OpenProjectHandler : ILspServiceNotificationHandler<OpenProjectHa
 {
     private readonly LanguageServerProjectSystem _projectSystem;
 
+    private readonly IAsynchronousOperationListener _listener;
+
     [ImportingConstructor]
     [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-    public OpenProjectHandler(LanguageServerProjectSystem projectSystem)
+    public OpenProjectHandler(LanguageServerProjectSystem projectSystem, IAsynchronousOperationListenerProvider listenerProvider)
     {
         _projectSystem = projectSystem;
+        _listener = listenerProvider.GetListener(FeatureAttribute.Workspace);
     }
 
     public bool MutatesSolutionState => false;
     public bool RequiresLSPSolution => false;
 
     Task INotificationHandler<NotificationParams, RequestContext>.HandleNotificationAsync(NotificationParams request, RequestContext requestContext, CancellationToken cancellationToken)
     {
+        using var token = _listener.BeginAsyncOperation(nameof(OpenProjectHandler));
```

Comment:
This will get completed immediately once the token completes, so either don't do the using (and call .CompletesToken on the task below) or make this async.